src/rpc/daemon_handler.cpp

```diff
@@ -0,0 +1,726 @@
+
+    // std::list is annoying
+    std::vector<cryptonote::transaction> found_txs_vec
+    {
+      std::make_move_iterator(std::begin(found_txs)),
+      std::make_move_iterator(std::end(found_txs))
+    };
+
+    std::vector<crypto::hash> missed_vec
+    {
+      std::make_move_iterator(std::begin(missed_hashes)),
+      std::make_move_iterator(std::end(missed_hashes))
+    };
+
+    std::vector<uint64_t> heights(num_found);
+    std::vector<bool> in_pool(num_found, false);
+    std::vector<crypto::hash> found_hashes(num_found);
+
+    for (size_t i=0; i < num_found; i++)
+    {
+      found_hashes[i] = get_transaction_hash(found_txs_vec[i]);
+      heights[i] = m_core.get_blockchain_storage().get_db().get_tx_block_height(found_hashes[i]);
+    }
+
+    // if any missing from blockchain, check in tx pool
+    if (!missed_vec.empty())
+    {
+      std::list<cryptonote::transaction> pool_txs;
+
+      m_core.get_pool_transactions(pool_txs);
+
+      for (auto& tx : pool_txs)
+      {
+        crypto::hash h = get_transaction_hash(tx);
+
+        auto itr = std::find(missed_vec.begin(), missed_vec.end(), h);
+
+        if (itr != missed_vec.begin())
+        {
+          found_hashes.push_back(h);
+          found_txs_vec.push_back(tx);
+          heights.push_back(std::numeric_limits<uint64_t>::max());
+          in_pool.push_back(true);
+          missed_vec.erase(itr);
+        }
+      }
+    }
+
+    for (size_t i=0; i < found_hashes.size(); i++)
+    {
+      cryptonote::rpc::transaction_info info;
+      info.height = heights[i];
+      info.in_pool = in_pool[i];
+      info.transaction = std::move(found_txs_vec[i]);
+
+      res.txs.emplace(found_hashes[i], std::move(info));
+    }
+                                      
+    res.missed_hashes = std::move(missed_vec);
+  }
+
+  void DaemonHandler::handle(KeyImagesSpent::Request& req, KeyImagesSpent::Response& res)
+  {
+    res.spent_status.resize(req.key_images.size(), KeyImagesSpent::STATUS::UNSPENT);
+
+    std::vector<bool> chain_spent_status;
+    std::vector<bool> pool_spent_status;
+
+    m_core.are_key_images_spent(req.key_images, chain_spent_status);
+    m_core.get_pool().have_key_images_as_spent(req.key_images, pool_spent_status);
+
+    for(uint64_t i=0; i < req.key_images.size(); i++)
+    {
+      if ( chain_spent_status[i] )
+      {
+        res.spent_status[i] = KeyImagesSpent::STATUS::SPENT_IN_BLOCKCHAIN;
+      }
+      else if ( pool_spent_status[i] )
+      {
+        res.spent_status[i] = KeyImagesSpent::STATUS::SPENT_IN_POOL;
+      }
+    }
+  }
+
+  void DaemonHandler::handle(GetTxGlobalOutputIndices::Request& req, GetTxGlobalOutputIndices::Response& res)
+  {
+    if (!m_core.get_tx_outputs_gindexs(req.tx_hash, res.output_indices))
+    {
+      res.status = Message::STATUS_FAILED;
+      res.error_details = "core::get_tx_outputs_gindexs() returned false";
+      return;
+    }
+
+    res.status = Message::STATUS_OK;
+
+  }
+
+  //TODO: handle "restricted" RPC
```

Comment:
Seems I can't make an "issue" on the branch to remind me to deal with this.  I wanna get the minimum feature set working for sure before I deal with this.